cmd/register/register_util.go

```diff
@@ -194,6 +191,138 @@ func hydrateSpec(message proto.Message) error {
 	return nil
 }
 
+func DownloadFileFromHTTP(ctx context.Context, ref storage.DataReference) (io.ReadCloser, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, ref.String(), nil)
+	if err != nil {
+		logger.Errorf(ctx, "failed to create new http request with context, %s", err)
+		return nil, err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, nil
+}
+
+/*
+Get file list from the args list.
+If the archive flag is on then download the archives to temp directory and extract it.
+The o/p of this function would be sorted list of the file locations.
+*/
+func getSortedFileList(ctx context.Context, args []string) ([]string, string, error) {
+	if !filesConfig.archive {
+		sort.Strings(args)
```

Comment:
Sorting is not necessary right? I guess its a good idea, but just trying to understand

Reply:
Its required in case of the archive since order in which files are archived seems to be different than the listing order .
In case of non-archived ones, if user passes in file2 file1 then this is just making sure we register to flyte in sorted order.

This is actually inline with how its done in flytekit aswell https://github.com/flyteorg/flytekit/blob/b6f81d3724787640db6ef99ecfddcdab074d2a83/flytekit/clis/flyte_cli/main.py#L1753
Let me know what you think

Reply:
I think this probably deserves a comment and hence is implicitly part of the contract